rumqttd/src/router/routing.rs

```diff
@@ -304,23 +303,32 @@ impl Router {
         // Retrieve previous connection state from graveyard
         let saved = self.graveyard.retrieve(&client_id);
         let clean_session = connection.clean;
-        let previous_session = saved.is_some();
+        let previous_session = saved.as_ref().is_some_and(|s| s.session_state.is_some());
         // for qos2 pending pubrels
         let mut pending_acks = VecDeque::new();
-        let tracker = if !clean_session {
-            let saved = saved.map_or(SavedState::new(client_id.clone()), |s| s);
-            connection.subscriptions = saved.subscriptions;
-            connection.events = saved.metrics;
-            // for using in acklog
-            pending_acks = saved.unacked_pubrels.clone();
-            outgoing.unacked_pubrels = saved.unacked_pubrels;
-            saved.tracker
-        } else {
-            // Only retrieve metrics in clean session
-            let saved = saved.map_or(SavedState::new(client_id.clone()), |s| s);
-            connection.events = saved.metrics;
-            Tracker::new(client_id.clone())
+
+        let tracker = 'tracker: {
+            if !clean_session {
+                let Some(saved) = saved else {
+                    break 'tracker Tracker::new(client_id.clone());
+                };
+                connection.events = saved.metrics;
+
+                let Some(session_state) = saved.session_state else {
+                    break 'tracker Tracker::new(client_id.clone());
+                };
```

Comment:
can we directly return `Tracker::new()` without using break with `'tracker` here?

Reply:
can't use `return` as it will directly return from the function! We want tracker in the `tracker` variable as result of the expression.

Otherwise, we would have to rely in way too much nested if-let-else blocks.

Reply:
tbh, i really loved the `'labels` haha, but updated code to not use em. as result, we have to use `map_or_else` to avoid crazy nesting of blocks :)